app/src/test/kotlin/compiler/intermediate/ControlFlowComparisonUtils.kt

```diff
@@ -0,0 +1,144 @@
+package compiler.intermediate
+
+import compiler.utils.Ref
+import compiler.utils.mutableKeyRefMapOf
+import kotlin.test.assertEquals
+import kotlin.test.assertSame
+import kotlin.test.assertTrue
+
+private fun mapLinkType(list: List<Pair<IFTNode, CFGLinkType>?>, type: CFGLinkType) = list.map { it?.copy(second = type) }
+
+infix fun ControlFlowGraph.hasSameStructureAs(cfg: ControlFlowGraph) {
+    val registersMap = mutableKeyRefMapOf<Register, Register>()
+    val callResultsMap = mutableKeyRefMapOf<IFTNode.DummyCallResult, IFTNode.DummyCallResult>()
+    val nodeMap = mutableKeyRefMapOf<IFTNode, IFTNode>()
+
+    fun <T> MutableMap<Ref<T>, T>.ensurePairSymmetrical(a: T, b: T) {
+        if (!this.containsKey(Ref(a))) {
+            this[Ref(a)] = b
+        }
+        assertSame(this[Ref(a)]!!, b)
+    }
+
+    infix fun IFTNode.hasSameStructureAs(iftNode: IFTNode) {
```

Comment:
Since this method now does assertions instead of returning bool, maybe rename it to `assertHasSameStructureAs` or `hasSameStructureAssertedAs`